website/content/docs/secrets/databases/redis.mdx

```diff
@@ -0,0 +1,124 @@
+---
+layout: docs
+page_title: Redis - Database - Secrets Engines
+description: |-
+  Redis is one of the supported plugins for the database secrets engine.
+  This plugin generates database credentials dynamically based on configured
+  roles for the Redis database.
+---
+
+# Redis Database Secrets Engine
+
+Redis is one of the supported plugins for the database secrets engine. This
+plugin generates database credentials dynamically based on configured roles for
+the Redis database.
+
+See the [database secrets engine](/docs/secrets/databases) docs for
+more information about setting up the database secrets engine.
+
+## Capabilities
+
+| Plugin Name                 | Root Credential Rotation | Dynamic Roles | Static Roles | Username Customization |
+| --------------------------- | ------------------------ | ------------- | ------------ | ---------------------- |
+| `redis-database-plugin`     | Yes                      | Yes           | Yes          | No                     |
+
+## Setup
+
+1.  Enable the database secrets engine if it is not already enabled:
+
+    ```shell-session
+    $ vault secrets enable database
+    Success! Enabled the database secrets engine at: database/
+    ```
+
+    By default, the secrets engine will enable at the name of the engine. To
+    enable the secrets engine at a different path, use the `-path` argument.
+
+1.  Configure Vault with the proper plugin and connection configuration:
+
+    ```shell-session
+    $ vault write database/config/my-redis-database \
+      plugin_name="redis-database-plugin" \
+      host="localhost" \
+      port=6379 \
+      username="vaultuser" \
+      password="vaultpass" \
+      allowed_roles="my-*-role"
+    ```
+
+1.  You should consider rotating the admin password. Note that if you do, the
+    new password will never be made available through Vault, so you should
+    create a Vault-specific database admin user for this.
+
+    ```shell-session
+    vault write -force database/rotate-root/my-redis-database
+    ```
+
+## Usage
+
+After the secrets engine is configured, configure dynamic and static roles
+to enable generating credentials.
+
+### Dynamic roles
+
+1.  Configure a dynamic role that maps a name in Vault to a JSON string
+    containing the Redis ACL rules which are documented [here](https://redis.io/commands/acl-cat) or in the output
+    of the `ACL CAT` redis command.
+
+    ```shell-session
+    $ vault write database/roles/my-dynamic-role \
+        db_name="my-redis-database" \
+        creation_statements='["+@admin"]' \
```

Comment:
Should we give an example that shows more options and good practices?

Something like '["~*", "-@admin", "+@read"]' to show how to pass multiple statements.